src/private.rs

```diff
@@ -72,100 +61,118 @@ impl RedactFlags {
                 if char.is_alphanumeric() {
                     if prefix_gas > 0 {
                         prefix_gas -= 1;
-                        redacted.push(char);
+                        fmt.write_char(char)?;
                     } else if middle_gas > 0 {
                         middle_gas -= 1;
-                        redacted.push(self.redact_char);
+                        fmt.write_char(self.redact_char)?;
                     } else {
-                        redacted.push(char);
+                        fmt.write_char(char)?;
                     }
                 } else {
-                    redacted.push(char);
+                    fmt.write_char(char)?;
                 }
             }
         }
+        Ok(())
     }
 
-    fn redact_full(&self, str: &str, redacted: &mut String) {
+    pub(crate) fn redact_full(&self, fmt: &mut std::fmt::Formatter, str: &str) -> std::fmt::Result {
         for char in str.chars() {
             if char.is_whitespace() || !char.is_alphanumeric() {
```

Comment:
This is the first time I've seen anyone use `is_alphanumeric` over `is_ascii_alphanumeric` correctly

Reply:
Unicode slay